ClipboardZanager/Sources/ClipboardZanager.Core.Desktop/Services/DataService.cs

```diff
@@ -306,14 +308,48 @@ internal List<DataIdentifier> GetDataIdentifiers(string[] formats)
             Requires.NotNull(foregroundWindow, nameof(foregroundWindow));
 
             var shouldSynchronize = true;
-            
+
+            BitmapImage icon = foregroundWindow.Icon;
```

Comment:
Why did you placed this code here? Can't you move this code to the place where we detect the icon the first time?